### PR TITLE
Fix create change hearing disposition workflow

### PR DESCRIPTION
**Pull request by ferristseng**
### Description

  * Fixes a bug in the change hearing disposition workflow, caused by an interaction between the hearing, and hearing task.

    https://github.com/department-of-veterans-affairs/caseflow/pull/12195 changed some code to pull the `hearing_task` from the most recently closed `hearing`. Unfortunately, that change does not work for cases where the `hearing` has been postponed because the `hearing_task` is only visible to the `hearing` if the `hearing_task` is not cancelled:

    https://github.com/department-of-veterans-affairs/caseflow/blob/d8006c672f6375e8ece66f57112820f442a981b8/app/models/concerns/has_hearing_task.rb#L6-L10

    This PR updates the code to find the `hearing` from the most recently closed `hearing_task`, which is how the code worked previously.

  * Display error message on `CreateChangeHearingDispositionTaskModal`
  * Move `create_change_hearing_disposition_task` updates into a transaction block

### CODE REVIEW

File: client/app/queue/components/QueueFlowModal.jsx [hunk 1]
```diff
@@ -81,8 +81,7 @@ QueueFlowModal.propTypes = {
   pathAfterSubmit: PropTypes.string,
   // submit should return a promise on which .then() can be called
   submit: PropTypes.func,
-  validateForm: PropTypes.func,
-  reloadPageAfterSubmit: PropTypes.bool
+  validateForm: PropTypes.func
```
Comment (ferristseng) [inline client/app/queue/components/QueueFlowModal.jsx:84]: Not used anywhere in the file

File: client/app/queue/CreateChangeHearingDispositionTaskModal.jsx [hunk 2]
```diff
@@ -54,10 +46,11 @@ class CreateChangeHearingDispositionTaskModal extends React.Component {
 
     return this.props.requestSave(`/tasks/${task.taskId}/request_hearing_disposition_change`, payload, successMsg).
       then((resp) => {
-        this.props.onReceiveAmaTasks(resp.body.tasks.data);
+        this.props.onReceiveAmaTasks(resp.body.tasks);
```
Comment (tomas-nava) [inline client/app/queue/CreateChangeHearingDispositionTaskModal.jsx:49]: why remove `.data` here?

File: client/app/queue/CreateChangeHearingDispositionTaskModal.jsx [hunk 3]
```diff
@@ -66,16 +59,20 @@ class CreateChangeHearingDispositionTaskModal extends React.Component {
...
-      pathAfterSubmit = "/queue"
+      pathAfterSubmit="/queue"
       submit={this.submit}
       validateForm={this.validateForm}
     >
+      {error &&
+        <Alert title={error.title} type="error">
+          {error.detail}
+        </Alert>
+      }
```
Comment (tomas-nava) [inline client/app/queue/CreateChangeHearingDispositionTaskModal.jsx:74]: Any way we could test for this?

File: client/app/queue/CreateChangeHearingDispositionTaskModal.jsx (hunk 2 above)
Comment (ferristseng) [reply client/app/queue/CreateChangeHearingDispositionTaskModal.jsx:49]: For this route, the data was coming back `{ "tasks": [ ... ]  }` -- I think this code was probably copied from somewhere else.

File: client/app/queue/CreateChangeHearingDispositionTaskModal.jsx (hunk 3 above)
Comment (ferristseng) [reply client/app/queue/CreateChangeHearingDispositionTaskModal.jsx:74]: I can't think of a great way to test this, but it was displaying the error when I was trying to reproduce the issue.

**Pull request merged**